p2p/host/basic/basic_host.go

```diff
@@ -53,6 +54,8 @@ var (
 	DefaultAddrsFactory = func(addrs []ma.Multiaddr) []ma.Multiaddr { return addrs }
 )
 
+const maxPeerRecordSize = 4096 // 4k to be compatible with rust-libp2p identify implementation
```

Comment:
I don't think we need this because the rust-libp2p identify implementation isn't even aware of signed peer records: https://github.com/libp2p/rust-libp2p/blob/master/protocols/identify/src/generated/structs.proto#L1

Reply:
I've changed it to 8k instead.